src/main/java/gov/nih/ncats/molwitch/cdk/CdkChemicalImpl.java

```diff
@@ -187,21 +214,25 @@ public class CdkChemicalImpl implements ChemicalImpl<CdkChemicalImpl>{
 		
     }
 
-    CachedSupplier<Void> perceiveAtomTypesOfNonQueryAtoms = CachedSupplier.of(()->{
+	CachedSupplier<Boolean> complexitySupplier =CachedSupplier.of(()->{
+		Logger.getLogger(this.getClass().getName()).info("getSizeOfLargestRingSystem(this): " + getSizeOfLargestRingSystem(this));
```

Comment:
Remove this line, or compute once and reference the field. Right now the getSizeoFLargestRingSystem is called twice